### ISSUE TITLE
Model training diverges after some level ?

### ISSUE THREAD

**Issue by erogol** [fchollet/keras#21]
Here my training output with Sequential model. As you can see model diverges after epoch 10. Any ideas about the reason?

Epoch 0
61878/61878 [==============================] - 5s - loss: 1.1788     
Epoch 1
61878/61878 [==============================] - 5s - loss: 1.0403     
Epoch 2
61878/61878 [==============================] - 5s - loss: 0.9919     
Epoch 3
61878/61878 [==============================] - 5s - loss: 0.9397     
Epoch 4
61878/61878 [==============================] - 5s - loss: 0.8915     
Epoch 5
61878/61878 [==============================] - 4s - loss: 0.8484     
Epoch 6
61878/61878 [==============================] - 5s - loss: 0.8145     
Epoch 7
61878/61878 [==============================] - 5s - loss: 0.7909     
Epoch 8
61878/61878 [==============================] - 4s - loss: 0.7627     
Epoch 9
61878/61878 [==============================] - 5s - loss: 0.7407     
Epoch 10
61878/61878 [==============================] - 6s - loss: 13.3614     
Epoch 11
61878/61878 [==============================] - 5s - loss: 26.6396     
Epoch 12
61878/61878 [==============================] - 5s - loss: 26.6453     
Epoch 13
61878/61878 [==============================] - 5s - loss: 26.6462     
Epoch 14
61878/61878 [==============================] - 5s - loss: 26.6461     
Epoch 15
61878/61878 [==============================] - 5s - loss: 26.6470     
Epoch 16
61878/61878 [==============================] - 5s - loss: 26.6468     
Epoch 17
61878/61878 [==============================] - 5s - loss: 26.6465     
Epoch 18
61878/61878 [==============================] - 3s - loss: 26.6468     
Epoch 19
61878/61878 [==============================] - 3s - loss: 26.6469

**Comment by patyork** [comment]
Could you post the model? This may be a hyperparameter issue.

**Comment by fchollet** [comment]
Yes, could you post a script to repro this (I assume you used the Kaggle Otto challenge data). It looks like an exploding gradient, and there are many things that could trigger one (most likely a learning rate issue?). 

In general when faced with this type of problem, you should try different learning rates, different optimizers, and try to introduce regularization techniques such as L1 regularization (supported by all optimizers in Keras, BatchNormalization, etc.).

**Comment by erogol** [comment]
```
model = Sequential()
model.add(Dense(20, 64, init='uniform', activation='tanh'))
model.add(Dropout(0.5))
model.add(Dense(64, 64, init='uniform', activation='tanh'))
model.add(Dropout(0.5))
model.add(Dense(64, 1, init='uniform', activation='softmax')

sgd = SGD(lr=0.1, decay=1e-6, momentum=0.9, nesterov=True)
model.compile(loss='mean_squared_error', optimizer=sgd)
```

I cannt post whole the script since I overwrote it  but this is the same model you give as an example. Only I use the data from some where else.

However, the strange thing is, it goes smoothly to some degree but suddenly it diverges. It is the first time I observed such a thing. I also used Adagrad and it works nice but this is strange anyway.

**Comment by erogol** [comment]
Also I see that time to time for Adagrad , model stucks to very high loss from the beginning but it goes well for different runs.

**Comment by nagadomi** [comment]
I guess the learning rate becomes negative.
https://github.com/fchollet/keras/blob/master/keras/optimizers.py#L48

**Comment by erogol** [comment]
yeah perfectly rational then it should be corrected.
